server/routes/v1/tokenDiscovery.js

```diff
@@ -114,21 +123,45 @@ async function cacheTokensList() {
   }
 }
 
-if (!timerStarted && process.env.REACT_APP_ENVIRONMENT === 'mainnet') {
-  timerStarted = true;
-  cacheTokensList();
-  setInterval(() => cacheTokensList(), TIME_INTERVAL);
+function startCaching() {
+  if (process.env.REACT_APP_ENVIRONMENT === 'mainnet') {
+    cacheTokensList();
+    const intervalId = setInterval(() => cacheTokensList(), TIME_INTERVAL);
+    if (!IS_PROD_ENV) {
+      setTimeout(() => {
+        log.info('stopping caching tokens');
+        clearInterval(intervalId);
+      }, TIME_TO_TEST);
+    }
+  }
 }
 
+startCaching();
+
 function sleep(ms) {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
 
 module.exports = async (req, res) => {
+  // * * * * * * * * IMPORTANT * * * * * * * * * * * *
+  // Running the BQ query costs money. Don't let it run if you don't need to.
   log.info(`getting token discovery`);
   try {
+    // if it's been a while since caching happened in the non-prod envs,
+    // then restart the caching
+    // (needed because `startCaching` turns off caching after TIME_TO_TEST for non-prod envs)
+    if (
+      !IS_PROD_ENV &&
+      cachedTokensList.time != null &&
+      Date.now() - cachedTokensList.time > TIME_INTERVAL * 2
```

Comment:
Why is this twice the time interval instead of 1x?

Reply:
It's a sanity check, in case there's a race condition around `TIME_INTERVAL`